backend/inspirehep/disambiguation/tasks.py

```diff
@@ -218,15 +252,25 @@ def _get_not_disambiguated_authors(authors):
             yield author
 
 
+def match_author(author, record, match_ambiguous_stub_author=False):
+    matched_author_data = match_author_with_config(author, match_ambiguous_stub_author)
+    if not matched_author_data:
+        matched_author_data = match_literature_author(
+            author, record, match_ambiguous_stub_author
+        )
+
+    return matched_author_data
+
+
 def _disambiguate_authors(authors_to_disambiguate, record):
     updated_authors = []
     for author in authors_to_disambiguate:
         if author.get("curated_relation"):
             continue
         assigned_author_recid = None
-        matched_author_data = match_author(author)
+        matched_author_data = match_author(author, record)
         if not matched_author_data:
-            matched_author_data = match_literature_author(author, record)
+            matched_author_data = match_author(author, record, True)
```

Comment:
wouldn't it be better to avoid doing all the queries on ES twice, and instead remember all the encountered author record IDs in the first pass, to later check for `stub`ness in case of ambiguity? I think it would also make the logic more clear. Now it's pretty hard to understand what's happening.